src/gui/DatabaseIcons.cpp

```diff
@@ -60,6 +60,7 @@ QPixmap DatabaseIcons::icon(int index, IconSize size)
 {
     if (index < 0 || index >= count()) {
         qWarning("DatabaseIcons::icon: invalid icon index %d", index);
+        // TODO should the index default to (count() - 1) instead, like was previously done in KdbxXmlReader???
```

Comment:
@droidmonkey one of the readers was also detecting the invalid icon index situation, but in that case, it was defaulting to the last icon in the default icon set. Wondering which strategy is the best.

Reply:
We should set it to the default icon index.

Reply:
Made this change.